homeassistant/components/switch_as_x/__init__.py

```diff
@@ -0,0 +1,39 @@
+"""Component to wrap switch entities in entities of other domains."""
+from __future__ import annotations
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ENTITY_ID, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+DOMAIN = "switch_as_x"
+
+PLATFORMS: list[Platform] = [Platform.LIGHT]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up a config entry."""
+    registry = er.async_get(hass)
+    try:
+        er.async_validate_entity_id(registry, entry.options[CONF_ENTITY_ID])
+    except vol.Invalid:
+        # The entity is identified by an unknown entity registry ID
+        _LOGGER.error(
+            "Failed to setup light switch for unknown entity %s",
+            entry.options[CONF_ENTITY_ID],
+        )
+        return False
```

Comment:
I wonder if this check is too harsh as it also won't retry it. Let's raise `ConfigEntryNotReady` or just forward it and let it mark as unavailable (but then it's harder to debug for user).

Reply:
This only fails if the entity registry ID is unknown, which is a permanent error.
It will not fail if the entity is identified by an entity id which happens to not be in the state machine.